GraphAlgorithmsTests/LargestCliqueNaiveFinderTests.cs

```diff
@@ -0,0 +1,213 @@
+                new int[] { 1, 1, 1, 1, 1, 1, 1, 1, 0, 0 }
+            };
+            yield return new TestCaseData(graph, Enumerable.Range(0, 9).ToList());
+
+            // many equal cliques
+            graph = new Graph(6);
+            graph.AdjacencyMatrix = new int[][] {
+                new int[] { 0, 1, 1, 0, 1, 1 },
+                new int[] { 1, 0, 1, 1, 0, 1 },
+                new int[] { 1, 1, 0, 1, 1, 0 },
+                new int[] { 0, 1, 1, 0, 1, 1 },
+                new int[] { 1, 0, 1, 1, 0, 1 },
+                new int[] { 1, 1, 0, 1, 1, 0 },
+            };
+            yield return new TestCaseData(graph, new List<int> { 0, 1, 2 });
+
+            //
+            // DEF: MAX_CLIQUE is number of vertices from largest clique
+            // DEF: deg = deg + 1 when there are edges in both sides between vertices
+            // graphs with vertices 'v' that deg(v) >= MAX_CLIQUE, but don't expand the biggest clique
+            //
+
+            // 1) v = 3: 3 = deg(v) == MAX_CLIQUE = 3
+            graph = new Graph(5);
+            graph.AdjacencyMatrix = new int[][] {
+                new int[] { 0, 1, 1, 1, 0 },
+                new int[] { 1, 0, 1, 1, 0 },
+                new int[] { 1, 1, 0, 1, 0 },
+                new int[] { 1, 1, 0, 0, 1 },
+                new int[] { 0, 0, 0, 1, 0 },
+            };
+            yield return new TestCaseData(graph, new List<int> { 0, 1, 2 });
+
+            // 2) v = 3 : 4 = deg(v) > MAX_CLIQUE = 3
+            graph = new Graph(6);
+            graph.AdjacencyMatrix = new int[][] {
+                new int[] { 0, 1, 1, 1, 0, 0 },
+                new int[] { 1, 0, 1, 0, 0, 0 },
+                new int[] { 1, 1, 0, 1, 0, 0 },
+                new int[] { 1, 0, 1, 0, 1, 1 },
+                new int[] { 0, 0, 0, 1, 0, 0 },
+                new int[] { 0, 0, 0, 1, 0, 0 },
+            };
+            yield return new TestCaseData(graph, new List<int> { 0, 1, 2 });
+
+            // some random graphs, graphs in which vertices 'v' that don't belong to clique have
+            // deg(v) < MAX_CLIQUE
+            var random = new Random();
+            int maxSize = 25;
+            for (int i = 0; i < 100; ++i)
+            {
+                graph = new Graph(random.Next(1, maxSize));
+                int cliqueSize = random.Next(1, graph.Size);
+                var clique = GenerateClique(graph, cliqueSize);
+                GenerateAdditionalVertices(graph, clique);
+                yield return new TestCaseData(graph, clique);
+            }
+        }
+
+        private static List<int> GenerateClique(Graph graph, int cliqueSize)
+        {
+            if (cliqueSize == 1)
+                return new List<int>() { 0 };
+            Random random = new Random();
+            List<int> clique = new List<int>();
+            while (clique.Count < cliqueSize)
+            {
+                int u = random.Next(graph.Size);
+                if (clique.Contains(u))
```

Comment:
I feel like we're pushing our luck here a little bit, if `random.Next()` can return stuff already in clique very often if `graph.Size` is close to `cliqueSize`

Reply:
Ok, now method randomizes vertices only from those available.